### PR TITLE
Implement support for boss level `A Most Unfortunate Malfunction`

### PR DESCRIPTION
**Pull request by jbzdarkid**

### CODE REVIEW

File: tests/test_data.py [hunk 1]
```diff
@@ -6347,6 +6347,337 @@
 PIPE:0,4,1
 PIPE:1,4,2""",
     r"""SOLUTION:Continuous Flow Test,Zig,4-0-0""",
+
+    r'''SOLUTION:Exploding Head Syndrome,vpumeyyv,4106-3-36,success
```
Comment (zig1000) [inline tests/test_data.py:6351]: Remove

File: tests/test_data.py [hunk 2]
```diff
@@ -8993,3 +9324,42 @@
 PIPE:1,14,-4
 ''',
 ]
+
+
+death_solutions = [
+r'''SOLUTION:Exploding Head Syndrome,darkid,6979-0-0,Death
```
Comment (zig1000) [inline tests/test_data.py:9330]: Remove

Comment (jbzdarkid) [reply tests/test_data.py:9330]: I can't quite tell what you're commenting on, is this trailing whitespace?

Comment (jbzdarkid) [reply tests/test_data.py:9330]: I cleaned up trailing whitespace, are you asking me to not check in additional tests?

There currently isn't any test coverage for DeathErrors, and it's particuarly valuable for boss levels like AMUM where the boss speed depends on player actions. This test was pretty instrumental to getting the timings right.

Comment (zig1000) [reply tests/test_data.py:9330]: No my bad, I was mixing up boss level names, thought you had left a world 4 boss test in the world 3 boss PR

Comment (jbzdarkid) [reply tests/test_data.py:9330]: Ah, fair enough -- I added an extra test in here from the zone 5 boss just so we'd have it. In theory we should have one for all bosses, but lazy ._.

**Comment by jbzdarkid** [comment]
Reminder from 12345ieee -- please don't forget to squash-merge this PR, to keep the history clean.

**Comment by zig1000** [comment]
Looks okay to me

**Pull request merged**